ksql-engine/src/main/java/io/confluent/ksql/planner/plan/StructuredDataSourceNode.java

```diff
@@ -332,13 +341,20 @@ private Integer findMatchingTimestampField(final String timestampFieldName) {
       final ValueMapperWithKey<K, GenericRow, GenericRow> valueMapper,
       final Serde<K> keySerde,
       final Serde<GenericRow> genericRowSerde,
-      final Serde<GenericRow> genericRowSerdeAfterRead
+      final Serde<GenericRow> genericRowSerdeAfterRead,
+      final KsqlConfig ksqlConfig,
+      final String opName
```

Comment:
is opName not always `getSourceOpName`?? Could you not just remove the param and replace with method call?